### ISSUE TITLE
how to use customized loss function and modify trainning procedure in caffe

### ISSUE THREAD

**Issue by ZMaple** [BVLC/caffe#962]
May i know if it is possible to use customized loss function in caffe? Also, if I want to change the formula of weight update in the backpropagation, is this allowed in caffe? Could you please give some hints about this? Thanks a lot!
